### PR TITLE
feat(docs): add blog post on COPY

### PR DESCRIPTION
**Pull request by amyshwang**

### CODE REVIEW

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+optimized import, c6a.4xlarge, gp2 volume - 17401 seconds
+optimized import, c5d.4xlarge, gp3 volume only - 1594 seconds
+optimized import, c5d.4xlarge, NVMe SSD for input file - 403 seconds
+-->
+
+The very first result on the above chart stands for the scenario we already discussed: c6a.4xlarge instance with the same 16 vCPU and 32GB RAM having a slow gp2 volume. We are including it to show the importance of the disk to the optimized import performance.
+
+Next, the gp3 volume only result doesn't use the local SSD, but manages to ingest the data into a partitioned table a lot faster than the gp2 run, thanks to the faster EBS volume. Finally, in the NVMe SSD run, the import takes less than 7 minutes - an impressive ingestion rate of 2.5Bln row/s (or 193MB/s) demonstrated without having the whole input file in the OS page cache. Here, the SSD is used as a read-only storage for the CSV file, while the database files are placed on the EBS volume. This is a convenient approach for a single-time import of high volume of data. As soon as the import is done, the SSD is no longer needed, so the EBS volume may be attached to a more affordable instance where the database would run.
```
Comment (amyshwang) [inline blog/2022-09-12-importing-3m-rows-with-io-uring.md:103]:
```suggestion
Next, the-gp3-volume-only result doesn't use the local SSD, but manages to ingest the data into a partitioned table a lot faster than the gp2 run, thanks to the faster EBS volume. Finally, in the NVMe SSD run, the import takes less than 7 minutes - an impressive ingestion rate of 2.5Bln row/s (or 193MB/s) without having the whole input file in the OS page cache. Here, the SSD is used as a read-only storage for the CSV file, while the database files are placed on the EBS volume. This is a convenient approach for a single-time import of high volume of data. As soon as the import is done, the SSD is no longer needed, so the EBS volume may be attached to a more affordable instance where the database would run.
```

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 2]
```diff
@@ -65,39 +65,46 @@ cairo.sql.copy.root=\home\my-user\my-qdb-import
...
 
-This improvement makes a terrific difference for anyone who wants to import their time series data to QuestDB, but also takes us close to the ClickHouse's results from the practical perspective. Another nice property of QuestDB's import is that, as soon as the import ends, the data is laid out on disk optimally, i.e. the column files are organized in partitions and no background merging is required.
-As we promised, we're going to explain why huge amount of RAM or a locally attached SSD make such a difference for QuestDB's import performance. To learn that, we're taking a leap into an engineering story full of trial and error.
+Now, as we promised, we're going to explain why huge amount of RAM or a locally attached SSD make such a difference for QuestDB's import performance. To learn that, we're taking a leap into an engineering story full of trial and error.
```
Comment (amyshwang) [inline blog/2022-09-12-importing-3m-rows-with-io-uring.md:107]:
```suggestion
Now, as we promised, we're going to explain why huge amount of RAM or a locally-attached SSD makes such a difference for QuestDB's import performance. To learn that, we're taking a leap into an engineering story full of trial and error.
```

**Review by amyshwang — COMMENTED** [review]
Minor editorial changes.
Paragraphs analysing the charts should be reviewed again once the charts are added, to ensure that the text corresponds to the charts.

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+We also did a run on the c6a.4xlarge instance (16 vCPU and 32GB RAM) from the original benchmark which is noticeably less powerful than the c6a.metal instance (192 vCPU and 384GB RAM). Yet, both instances had a rather slow gp2 500GB EBS volume. The result was 17,401 seconds for the optimized import. So, in spite of a very slow disk, c6a.metal is 52x faster than c6a.4xlarge. Why is that? The answer is simple. The metal instance has a huge amount of memory, so once the CSV file gets decompressed, it fully fits into the OS page cache. Hence, the import doesn't do any physical reads for the input file and instead reads the pages from the memory (note: the machine is a [NUMA](https://en.wikipedia.org/wiki/Non-uniform_memory_access) one, but non-local memory access is still way faster than the disk reads). That's why we observe such huge difference here for QuestDB and, also, you may notice a 2.5x difference for ClickHouse in the original benchmark.
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply blog/2022-09-12-importing-3m-rows-with-io-uring.md]: Addressed in 9c98657077bd964fdbfc399c93c30e5011811812

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+What we got is the following:
+
+<!--TODO: create a chart
+optimized import, c6a.4xlarge, gp2 volume - 17401 seconds
+optimized import, c5d.4xlarge, gp3 volume only - 1594 seconds
+optimized import, c5d.4xlarge, NVMe SSD for input file - 403 seconds
+-->
+
+The very first result on the above chart stands for the scenario we already discussed: c6a.4xlarge instance with the same 16 vCPU and 32GB RAM having a slow gp2 volume. We are including it to show the importance of the disk to the optimized import performance.
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply blog/2022-09-12-importing-3m-rows-with-io-uring.md]: Addressed in 9c98657077bd964fdbfc399c93c30e5011811812

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+
+<Screenshot
+  alt="Bar chart showing QuestDB import performance using different HW."
+  height={360}
+  src="/img/blog/2022-09-12/questdb_comparison.png"
+  width={650}
+/>
+
+The very last result on the above chart stands for the scenario we already
+discussed: c6a.4xlarge instance with the same 16 vCPU and 32GB RAM having a slow
```
Comment (amyshwang) [inline blog/2022-09-12-importing-3m-rows-with-io-uring.md:176]:
```suggestion
discussed: c5d.4xlarge instance with the same 16 vCPU and 32GB RAM having a slow
```

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+  alt="Bar chart showing import comparison. From fast to slow: ClickHouse, QuestDB, Apache Pinot, Timescale, DuckDB, and Apache Druid."
+  height={360}
+  src="/img/blog/2022-09-12/banner.png"
+  width={650}
+/>
+
+In this blog post, QuestDB’s very own
+[Andrei Pechkurov](https://github.com/puzpuzpuz) presents the newly enhanced SQL
+[`COPY`](https://questdb.io/docs/reference/sql/copy) statement and takes us
+through the journey of benchmarking. As our Core Engineer, Andrei also shares
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply blog/2022-09-12-importing-3m-rows-with-io-uring.md:29]: Tried to apply the suggestion in 01d62e69fcf0346a46daae920dd195222c2d042b

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+
+Honestly speaking, we find the choice of the metal instance in the ClickBench
+results rather synthetic, as it makes little sense to use a very powerful (and
+expensive) machine in combination with a very slow (and cheap) disk. So, we did
+a benchmark run on a different test stand:
+
+- c5d.4xlarge EC2 instance (16 vCPU and 32GB RAM), Amazon Linux 2 with 5.15.50
+  kernel
+- 400GB NVMe drive
+- 250GB gp3, 16K IOPS and 1GB/s throughput
```
Comment (amyshwang) [inline blog/2022-09-12-importing-3m-rows-with-io-uring.md:160]: Let's make this list correspond to the diagram, also making it more relevant to the chart:

```suggestion
- NVMe SSD: 400GB NVMe drive
- gp3: 250GB gp3, 16K IOPS and 1GB/s throughput
- gp2: EC2 instance (16 vCPU and 32GB RAM), Amazon Linux 2 with 5.15.50
  kernel


```

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+  alt="Bar chart showing QuestDB import performance using different HW."
+  title="QuestDB ingestion time for ClickBench's 76GB CSV file by instance type and storage."
+  height={360}
+  src="/img/blog/2022-09-12/questdb-comparison.png"
+  width={650}
+/>
+
+The very last result on the above chart stands for the scenario we already
+discussed: c5d.4xlarge instance with a slow gp2 volume. We are including it to
+show the importance of the disk speed to the performance.
```
Comment (amyshwang) [inline blog/2022-09-12-importing-3m-rows-with-io-uring.md:174]: I don't think this has been discussed:
```suggestion
The very last result on the above chart stands for the scenario of c5d.4xlarge instance with a slow gp2 volume. We are including it to
show the importance of the disk speed to the performance.
```

**Review by amyshwang — COMMENTED** [review]
Hope my interpretation is right?

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md (hunk 8 above)
Comment (puzpuzpuz) [reply blog/2022-09-12-importing-3m-rows-with-io-uring.md]: Applied in 0791689dcbf7604b05b12099b897018057f05ba4

File: blog/2022-09-12-importing-3m-rows-with-io-uring.md (hunk 7 above)
Comment (puzpuzpuz) [reply blog/2022-09-12-importing-3m-rows-with-io-uring.md]: Changed the test in the spirit of the suggestion (but not exactly as proposed) in 0791689dcbf7604b05b12099b897018057f05ba4

**Review by puzpuzpuz — APPROVED** [review]

**Pull request merged**